common/djangoapps/third_party_auth/provider.py

```diff
@@ -84,16 +85,16 @@ def get_enabled_by_backend_name(cls, backend_name):
         """
         if backend_name in _PSA_OAUTH2_BACKENDS:
             provider = OAuth2ProviderConfig.current(backend_name)
-            if provider.enabled:
+            if provider.enabled_for_current_site:
```

Comment:
Update the docstring of this function to indicate you are returning a generator for over providers for the current site.